test/hotspot/jtreg/containers/docker/TestJcmd.java

```diff
@@ -169,7 +169,8 @@ private static Process startObservedContainer() throws Exception {
             .addDockerOpts("--name", CONTAINER_NAME)
             .addClassOptions("" + TIME_TO_RUN_CONTAINER_PROCESS);
 
-        if (IS_PODMAN) {
+        String uid = getId("-u");
+        if (IS_PODMAN && !ROOT_UID.equals(uid)) {
```

Comment:
Suggestion - to avoid execution in non-podman case:
```
if (IS_PODMAN && !ROOT_UID.equals(getId("-u"))) {
```
Alternatively in `generateCustomDockerfile` we could always check the uid and stash that away for later use here?

Reply:
Thanks, updated!